src/modules/bruteforce.rs

```diff
@@ -148,3 +157,76 @@ fn approx_color(c: colorgrad::Color) -> yansi::Color {
 
     yansi::Color::Fixed(16 + 36 * r + 6 * g + b)
 }
+
+fn gen_pass_and_hash<T: Rng>(rng: &mut T) -> (String, String) {
+    let pass = format!(
+        "{}{:02}",
+        FirstName(EN).fake::<&str>().to_lowercase(),
+        rng.gen_range(0..99)
+    );
+    let hash = sha256(&pass);
+    (pass, hash)
+}
+
+struct HashGuesser {
+    hash: Vec<u8>,
+    guesses: Vec<u8>,
+    progress: usize,
+    len: usize,
+    rng: ThreadRng,
+}
+
+impl HashGuesser {
+    fn new(hash: &str) -> Self {
+        // the hash string must only contains lowercase hex characters
+        debug_assert!(hash
+            .chars()
+            .all(|c| c.is_ascii_hexdigit() && !c.is_ascii_uppercase()));
+
+        Self {
+            hash: hash.bytes().collect(),
+            guesses: vec![], // will be set by run_guess
+            progress: 0,
+            len: hash.len(),
+            rng: rand::thread_rng(),
+        }
+    }
+
+    fn update_progress(&mut self) {
+        let progress = &mut self.progress;
```

Comment:
Do you need this temporary binding? I think you could just write to `self.progress` directly right?